sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/CreateJacksonParser.scala

```diff
@@ -60,4 +61,13 @@ private[sql] object CreateJacksonParser extends Serializable {
         jsonFactory.createParser(is)
     }
   }
+
+  def internalRow(
+    jsonFactory: JsonFactory,
+    row: InternalRow,
+    charset: Option[String] = None
+  ): JsonParser = {
+    val is = new ByteArrayInputStream(row.getBinary(0))
```

Comment:
it's weird to assume the data is in the first column of the input row. How about
```
def bytes(jsonFactory: JsonFactory, bytes: Array[Byte], charset: Option[String])
```